File: leet/taum.py

def taum(arr1,arr2):
	b,w = arr1
	x,y,z = arr2
	init_sum = b*x + w*y
	#print "initially: ",init_sum
	for i in range(1,b+1):
		new_sum = (b-i)*x + (w+i)*y + i*z
		if new_sum < init_sum:
			init_sum = new_sum
			#break;
	#print "finally: ",init_sum
	for j in range(1,w+1):
		new_sum = (b+j)*x + j*z + (w-j)*y
		if new_sum < init_sum:
			init_sum = new_sum
			#break;
		
	print init_sum

t = int(raw_input())
for i in range(t):
	dude1 = raw_input().split()
	dude1 = [int(x) for x in dude1]
	dude2 = raw_input().split()
	dude2 = [int(x) for x in dude2]
	taum(dude1,dude2)

File: leet/taum.cpp

#include <iostream>
#include <vector>
using namespace std;

long taum(vector<long> arr1, vector<long> arr2)
{
	long b = arr1[0];
	long w = arr1[1];
	long x = arr2[0];
	long y = arr2[1];
	long z = arr2[2];
	long init_sum = b*x + w*y;
	for (long i = 1; i<=b; i++)
	{
		long new_sum = (b-i)*x + (w+i)*y + i*z;
		if (new_sum < init_sum)
			init_sum = new_sum;
	}
	for (long i = 1; i<=w; i++)
	{
		long new_sum = (b+i)*x + (w-i)*y + i*z;
		if (new_sum < init_sum)
			init_sum = new_sum;
	}
	return init_sum;
}

int main()
{

	long t;
	cin >> t;
	for (long i = 0; i<t; i++)
	{
		vector<long> v1;
		long a,b,c,d,e;
		cin >> a >> b;
		v1.push_back(a);
		v1.push_back(b);
		vector<long> v2;
		cin >> c >> d >> e;
		v2.push_back(c);
		v2.push_back(d);
		v2.push_back(e);
		long ans;
		ans = taum(v1,v2);
		cout.setf(ios::fixed);
		cout << ans << endl;
	}
}
	


